info-creator/io500_info_editor.py

```diff
@@ -148,21 +149,33 @@ def process(data, schema, tokens):
     else:
       print("Error: cannot validate path (or value): %s" % token)
 
-def edit_infos(file, tokens, schemafile = "schema-io500.json"):
+def edit_infos(sitefile, tokens, schemafile = "schema-io500.json"):
   global units, templates, schema
+
+  check_requirements(sitefile, schemafile)
+
   with open(schemafile, 'r') as f:
     schema = json.load(f)
     templates = schema["SCHEMES"]
     units = schema["UNITS"]
 
-  with open(file, 'r+') as f:
+  with open(sitefile, 'r+') as f:
       data = json.load(f)
       process(data["DATA"], schema["SYSTEM"], tokens)
 
       f.seek(0)
       json.dump(data, f, indent=2)
       f.truncate()
 
+def check_requirements(site, schema = "schema-io500.json"):
+  if not os.path.exists(site):
+    import subprocess
+    print("Downloading new site specification")
```

Comment:
Preferably, this should prompt the user if the download is OK before doing it, when running with an interactive tty.

Reply:
Could do. We should do it consistently then. In prepare.sh, we just download (clone) repos and build stuff without user intervention.